### PR TITLE
T6698: firewall: add matcher for vlan type.

### PR DESCRIPTION
**Pull request by nicolas-fort**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6698

## Component(s) name

firewall
## Proposed changes

Extend matching capabilities for vlan header.

## How to test

Rules to match all traffic from certain vlan within a bridge, and then accept arp for such vlan:
```
# All interfaces from bridge 3:
set firewall group interface-group br3-ifaces interface 'eth3'
set firewall group interface-group br3-ifaces interface 'eth4'
set firewall group interface-group br3-ifaces interface 'eth5'
set firewall group interface-group br3-ifaces interface 'eth8'
# Forward rule:
set firewall bridge forward filter rule 310 action 'jump'
set firewall bridge forward filter rule 310 inbound-interface group 'br3-ifaces'
set firewall bridge forward filter rule 310 jump-target 'br3-fwd'
set firewall bridge forward filter rule 310 vlan id '35'
# And rules for accepting arp on such vlan:
set firewall bridge name br3-fwd rule 110 action 'accept'
set firewall bridge name br3-fwd rule 110 description 'Allow ARP'
set firewall bridge name br3-fwd rule 110 log
set firewall bridge name br3-fwd rule 110 vlan ethernet-type 'arp'
set firewall bridge name br3-fwd rule 110 vlan id '35'


```
And nftables:
```
table bridge vyos_filter {
        set I_br3-ifaces {
                type ifname
                flags interval
                auto-merge
                elements = { "eth3",
                             "eth4",
                             "eth5",
                             "eth8" }
        }

        chain VYOS_FORWARD_filter {
                type filter hook forward priority filter; policy accept;
                iifname @I_br3-ifaces vlan id 35 counter packets 0 bytes 0 jump NAME_br3-fwd comment "bri-FWD-filter-310"
...
        chain NAME_br3-fwd {
                vlan id 35 vlan type arp log prefix "[bri-NAM-br3-fwd-110-A]" counter packets 0 bytes 0 accept comment "bri-NAM-br3-fwd-110"
...
```

## Smoketest result

test_firewall.py --> OK

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by sever-sever — COMMENTED** [review]
Why did you change 0644 -> 0755 for the xml?

File: python/vyos/firewall.py [hunk 1]
```diff
@@ -496,6 +496,19 @@ def parse_rule(rule_conf, hook, fw_name, rule_id, ip_name):
             output.append(f'vlan id {rule_conf["vlan"]["id"]}')
         if 'priority' in rule_conf['vlan']:
             output.append(f'vlan pcp {rule_conf["vlan"]["priority"]}')
+        if 'ethernet_type' in rule_conf['vlan']:
```
Comment (sever-sever) [inline python/vyos/firewall.py:499]: Can we use `ether-type` without VLAN?

Comment (nicolas-fort) [reply python/vyos/firewall.py:499]: Yes, we can use both:
- Without vlan, added previously: https://github.com/vyos/vyos-1x/blob/current/python/vyos/firewall.py#L154-L166
- And now extending the feature in order to match if types when vlan header is present.

**Comment by nicolas-fort** [comment]
> Why did you change 0644 -> 0755 for the xml?

Looks like it was done automatically. I've reverted back permission to 644, and also change permissions for other files. Now all files under interface-definitions/include/firewall are set to 644.

**Review by c-po — APPROVED** [review]

**Review by sever-sever — APPROVED** [review]

**Comment by sever-sever** [comment]
@nicolas-fort let us know if we need to backport it or test it before backporting.
Thanks

**Pull request merged**